src/CMakeLists.txt

```diff
@@ -246,17 +230,151 @@ ADD_DEFINITIONS(-DEIGEN_MPL2_ONLY)
+  FIND_LIBRARY(CLP_LIBRARY NAMES Clp)
+  FIND_LIBRARY(CLPSOLVER_LIBRARY NAMES ClpSolver)
+  FIND_LIBRARY(OSICLP_LIBRARY NAMES OsiClp)
+  IF(NOT DEFINED CLP_LIBRARY AND NOT DEFINED CLP_SOLVER_LIBRARY AND NOT DEFINED OSICLP_LIBRARY)
+    SET(OpenMVG_USE_INTERNAL_CLP ON)
+  ELSE()
+    SET(CLP_LIBRARIES ${CLP_LIBRARY} ${CLPSOLVER_LIBRARY} ${OSICLP_LIBRARY})
+  ENDIF()
+ELSE(CLP_INCLUDE_DIRS)
+  SET(OpenMVG_USE_INTERNAL_CLP ON)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_CLP)
+  SET(CLP_INCLUDE_DIRS
+     ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/osi_clp/Clp/src/
+     ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/osi_clp/Clp/src/OsiClp/)
+  SET(CLP_LIBRARIES lib_clp lib_OsiClpSolver)
+ENDIF()
+
+# ==============================================================================
+# Osi
+# ==============================================================================
+# - internal by default (Osi),
+# - external if OSI_DIR_HINTS and a valid Osi setup is found
+# ==============================================================================
+FIND_PATH(OSI_INCLUDE_DIRS OsiConfig.h ${OSI_DIR_HINTS})
+IF(OSI_INCLUDE_DIRS)
+  FILE(READ "${OSI_INCLUDE_DIRS}/OsiConfig.h" OSI_CONFIG)
+  STRING(REGEX REPLACE ".*OSI_VERSION \"([0-9.]+)\".*" "\\1" OSI_VERSION ${OSI_CONFIG})
+  FIND_LIBRARY(OSI_LIBRARY NAMES Osi)
+  IF(NOT DEFINED OSI_LIBRARY)
+    SET(OpenMVG_USE_INTERNAL_OSI ON)
+  ENDIF()
+ELSE(OSI_INCLUDE_DIRS)
+  SET(OpenMVG_USE_INTERNAL_OSI ON)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_OSI)
+  SET(OSI_INCLUDE_DIRS
+     ${CMAKE_CURRENT_SOURCE_DIR}/dependencies/osi_clp/Osi/src/Osi/)
+  SET(OSI_LIBRARY lib_Osi)
+ENDIF()
+
+# ==============================================================================
+# Lemon
+# ==============================================================================
+# - internal by default (Lemon),
+# - external if LEMON_DIR_HINTS and a valid Lemon setup is found
+# ==============================================================================
+FIND_PATH(LEMON_INCLUDE_DIR config.h ${LEMON_DIR_HINTS})
+IF(LEMON_INCLUDE_DIR)
+  FILE(READ "${LEMON_INCLUDE_DIR}/config.h" LEMON_CONFIG)
+  STRING(REGEX REPLACE ".*LEMON_VERSION \"([0-9.]+)\".*" "\\1" LEMON_VERSION ${LEMON_CONFIG})
+  FIND_LIBRARY(LEMON_LIBRARY NAMES emon)
+  IF(NOT DEFINED LEMON_LIBRARY)
+    SET(OpenMVG_USE_INTERNAL_LEMON ON)
+  ELSE()
+    GET_FILENAME_COMPONENT(LEMON_INCLUDE_DIRS ${LEMON_INCLUDE_DIR} PATH)
+  ENDIF()
+ELSE(LEMON_INCLUDE_DIR)
+  SET(OpenMVG_USE_INTERNAL_LEMON ON)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_LEMON)
+  SET(LEMON_INCLUDE_DIRS
+     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/lemon
+     ${PROJECT_BINARY_DIR}/third_party/lemon)
+  SET(LEMON_LIBRARY lemon)
+ENDIF()
+
+# ==============================================================================
+# Internal CLP/OSI/COINUTILS libraries:
+# ==============================================================================
+ADD_SUBDIRECTORY(dependencies/osi_clp/)
```

Comment:
Hmm, github cannot patch recursive sub-projects ?!

This one below (screwed due to github) seems to be lost ! Should i open separate PR for openMVG-ThirdParty ?

---------------->>>>---------------
diff -Nru openMVG/src/dependencies/osi_clp/CMakeLists.txt openMVG-shared/src/dependencies/osi_clp/CMakeLists.txt
--- openMVG/src/dependencies/osi_clp/CMakeLists.txt     2015-07-23 02:33:27.000000000 +0300
+++ openMVG-shared/src/dependencies/osi_clp/CMakeLists.txt      2015-07-28 22:56:56.060799915 +0300
@@ -5,6 +5,15 @@
 # CMake version control
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 
+
+# static libraries only
+SET(BUILD_SHARED_LIBS OFF)
+
+# use PIC code for link into shared lib
+IF(UNIX)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+ENDIF(UNIX)
+

 INCLUDE_DIRECTORIES(.
   ./CoinUtils/src/
@@ -28,6 +37,20 @@
   ) 
 ENDIF(MSVC)
 
-ADD_SUBDIRECTORY(Osi)
-ADD_SUBDIRECTORY(Clp)
-ADD_SUBDIRECTORY(CoinUtils)
+
+IF(DEFINED OpenMVG_USE_INTERNAL_COINUTILS)
+  ADD_SUBDIRECTORY(CoinUtils)
+  SET_PROPERTY(TARGET lib_CoinUtils PROPERTY FOLDER OpenMVG/3rdParty/osi_clp)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_OSI)
+  ADD_SUBDIRECTORY(Osi)
+  SET_PROPERTY(TARGET lib_Osi PROPERTY FOLDER OpenMVG/3rdParty/osi_clp)
+ENDIF()
+
+IF(DEFINED OpenMVG_USE_INTERNAL_CLP)
+  ADD_SUBDIRECTORY(Clp)
+  SET_PROPERTY(TARGET lib_OsiClpSolver PROPERTY FOLDER OpenMVG/3rdParty/osi_clp)
+  SET_PROPERTY(TARGET lib_clp PROPERTY FOLDER OpenMVG/3rdParty/osi_clp)
+ENDIF()
+
----------------<<<<---------------